gui/web/src/utils/metricsTracker.js

```diff
@@ -0,0 +1,27 @@
+class MetricsTracker {
+    constructor(apiKey) {
+        this.apiKey = apiKey;
+        this.apiUrl = "https://api2.amplitude.com/2/httpapi";
+
+        // TODO: Implement more functions.
+        this.sendEvent = this.sendEvent.bind(this);
```

Comment:
should add this here too: `this._asyncRequests = {};`
the convention is to initialize all objects in the constructor so we know what all the objects are as opposed to finding new objects in a random function. this becomes more important as a given class grows.

Reply:
PS context about some code style: infrastructure variables for non-product related bookkeeping (engineering related) are prefixed with `_` like here. `_asyncRequests` is the only example I can think off in the frontend codebase.